MvvmCross/Platforms/Ios/Presenters/MvxIosViewPresenter.cs

```diff
@@ -410,10 +410,25 @@ private Task<bool> ShowPopoverViewControllerChild(UIViewController viewControlle
         {
             ValidateArguments(viewController, attribute);
 
-            // Content size should be set to a target view controller, not the navigation one
-            if (attribute.PreferredContentSize != default)
+            // Check if the requested viewmodel is already presented AND ignore this request
```

Comment:
This part looks a bit fishy.

Can you elaborate why it is necessary to dispose a ViewController here?

Which one is getting disposed?

Reply:
You are correct, this was actually a leftover I forgot to remove from when DidDismiss, instead of the now WillDismiss, was overridden in MvxModalPresentationControllerDelegate.

Now with moving the Remove call from the list and overriding WillDismiss instead of DidDismiss, I never stepped into the case where the modal viewcontroller to be shown was still present on the list, so I removed the entire if block.

Reply:
Alright, thanks :)